sdk/storage/Azure.Storage.Common/src/Shared/StorageClientOptions.cs

```diff
@@ -42,6 +42,24 @@ public static void Initialize(this ClientOptions options)
             new StorageSharedKeyPipelinePolicy(
                 credential ?? throw Errors.ArgumentNull(nameof(credential)));
 
+        /// <summary>
+        /// Get an authentication policy to sign Storage requests.
+        /// </summary>
+        /// <param name="credential">Credential to use.</param>
+        /// <param name="resourceUri">Resource Uri. Must not contain shared access signature.</param>
+        /// <returns>An authentication policy.</returns>
+        public static HttpPipelinePolicy AsPolicy<TUriBuilder>(this AzureSasCredential credential, Uri resourceUri)
+        {
+            Argument.AssertNotNull(resourceUri, nameof(resourceUri));
+            var queryParameters = resourceUri.GetQueryParameters();
+            if (queryParameters.ContainsKey("sig"))
+            {
+                throw Errors.SasCredentialRequiresUriWithoutSas<TUriBuilder>(resourceUri);
+            }
+            return new AzureSasCredentialSynchronousPolicy(
+                credential ?? throw Errors.ArgumentNull(nameof(credential)));
```

Comment:
I think it's better to check preconditions at the beginning of the method body.

Reply:
done.